```bash
mypy.....................................................................Failed
- hook id: mypy
- duration: 127.2s
- exit code: 1

pandas/tests/extension/test_period.py:95: error: Item "dtype[Any]" of "dtype[Any] | ExtensionDtype" has no attribute "freq"  [union-attr]
pandas/tests/extension/test_period.py:95: error: Item "ExtensionDtype" of "dtype[Any] | ExtensionDtype" has no attribute "freq"  [union-attr]
pandas/tests/extension/test_datetime.py:131: error: Item "dtype[Any]" of "dtype[Any] | ExtensionDtype" has no attribute "tz"  [union-attr]
pandas/tests/extension/test_datetime.py:131: error: Item "ExtensionDtype" of "dtype[Any] | ExtensionDtype" has no attribute "tz"  [union-attr]
Found 4 errors in 2 files (checked 1413 source files)
```

--- a/pandas/tests/extension/test_datetime.py
+++ b/pandas/tests/extension/test_datetime.py
@@ -128,7 +128,10 @@ class TestDatetimeArray(base.ExtensionTests):
             result = res_op(skipna=skipna)
             expected = exp_op(skipna=skipna)
             if op_name in ["mean", "median"]:
-                expected = pd.Timestamp(expected, tz=ser.dtype.tz)
+                # error: Item "dtype[Any]" of "dtype[Any] | ExtensionDtype"
+                # has no attribute "tz"
+                tz = ser.dtype.tz  # type: ignore[union-attr]
+                expected = pd.Timestamp(expected, tz=tz)
             else:
                 expected = pd.Timedelta(expected)
             tm.assert_almost_equal(result, expected)

--- a/pandas/tests/extension/test_period.py
+++ b/pandas/tests/extension/test_period.py
@@ -92,7 +92,10 @@ class TestPeriodArray(base.ExtensionTests):
             exp_op = getattr(alt, op_name)
             result = res_op(skipna=skipna)
             expected = exp_op(skipna=skipna)
-            expected = Period._from_ordinal(int(expected), freq=ser.dtype.freq)
+            # error: Item "dtype[Any]" of "dtype[Any] | ExtensionDtype" has no
+            # attribute "freq"
+            freq = ser.dtype.freq  # type: ignore[union-attr]
+            expected = Period._from_ordinal(int(expected), freq=freq)
             tm.assert_almost_equal(result, expected)
 
         else:

